### PR TITLE
Quarantine test and fix up conditional check

### PR DESCRIPTION
**Pull request by captainsafia**

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -15,7 +15,7 @@
     <AnalysisMode>Default</AnalysisMode>
   </PropertyGroup>
 
-  <PropertyGroup Condition=" '$(BuildingInsideVisualStudio)' == 'true' AND '$(VisualStudioVersion)' == '17.0' ">
+  <PropertyGroup Condition="'$(MSBuildRuntimeType)' != 'Core' AND $([MSBuild]::VersionLessThan('$(VisualStudioVersion)', '17.0'))">
```
Comment (dougbu) [inline Directory.Build.props:18]: You want
```suggestion
  <PropertyGroup Condition="'$(MSBuildRuntimeType)' != 'Core' AND $([MSBuild]::VersionEquals('$(VisualStudioVersion)', '17.0'))">
```

Comment (dougbu) [inline Directory.Build.props:18]: Or
```suggestion
  <PropertyGroup Condition="'$(MSBuildRuntimeType)' != 'Core' AND $([MSBuild]::VersionGreaterThanOrEquals('$(VisualStudioVersion)', '17.0'))">
```

File: eng/targets/CSharp.Common.targets [hunk 2]
```diff
@@ -36,7 +36,7 @@
 
     <!-- Enable .NET code style analysis during build for src projects. -->
     <!-- Workaround bug where turning this on produces warnings in VS -->
```
Comment (dougbu) [inline eng/targets/CSharp.Common.targets:38]: @pranavkm do we have an issue tracking turning this on unconditionally❔

File: eng/targets/CSharp.Common.targets [hunk 3]
```diff
@@ -36,7 +36,7 @@
...
-    <EnforceCodeStyleInBuild Condition="'$(BuildingInsideVisualStudio)' == 'true' AND '$(VisualStudioVersion)' &lt; '17.0'">false</EnforceCodeStyleInBuild>
+    <EnforceCodeStyleInBuild Condition="'$(MSBuildRuntimeType)' != 'Core' AND $([MSBuild]::VersionLessThan('$(VisualStudioVersion)', '17.0'))">false</EnforceCodeStyleInBuild>
```
Comment (dougbu) [inline eng/targets/CSharp.Common.targets:39]: And, was I correct we should enforce code style when building w/ desktop `msbuild`❔

File: eng/targets/CSharp.Common.targets (hunk 2 above)
Comment (pranavkm) [reply eng/targets/CSharp.Common.targets:38]: https://github.com/dotnet/roslyn/issues/54867

Comment (dougbu) [reply eng/targets/CSharp.Common.targets:38]: nit: might be worth adding that to this comment

now hidden:
> And, was I correct we should enforce code style when building w/ [newer] desktop `msbuild`❔

Comment (captainsafia) [reply eng/targets/CSharp.Common.targets:38]: Nope -- looks like the BuildInVS check is sufficient. Added hte code comment.

**Review by dougbu — APPROVED** [review]
WFM

**Pull request merged**